src/server/pkg/storage/fileset/index/index.proto

```diff
@@ -20,22 +19,13 @@ message Range {
   chunk.DataRef chunk_ref = 3;
 }
 
-message FileOp {
-  Op op = 1;
-  repeated DataOp data_ops = 2;
-  repeated chunk.DataRef data_refs = 3;
-}
-
-// TODO: Overwrite op is technically unecessary, but it might be better to be explicit.
-enum Op {
-  APPEND = 0;
-  DELETE = 1;
-  OVERWRITE = 2;
+message File {
+  repeated Part parts = 1;
+  repeated chunk.DataRef data_refs = 2;
```

Comment:
Why do we also need data_refs for a File?  I would prefer using a Part with an empty tag or something like that.

Reply:
I think this will get a bit cleaner if we end up moving to the parts only model (index per part rather than per file). With the current model, I think it is clearer to have the data refs for the full file associated with the file data structure rather than a special part. The code would get pretty wonky if that was how we managed the full file data refs.